File: sol/solution/0001-0100/0013. Roman to Integer/Solution.ex
defmodule Solution do
  def roman_to_int(s) do
    map = %{"I" => 1, "V" => 5, "X" => 10, "L" => 50, "C" => 100, "D" => 500, "M" => 1000}
    chars = String.graphemes(s)
    chars
    |> Enum.with_index()
    |> Enum.reduce(0, fn {char, i}, acc ->
      curr = map[char]
      next = if i + 1 < length(chars), do: map[Enum.at(chars, i + 1)], else: 0
      if curr < next, do: acc - curr, else: acc + curr
    end)
  end
end

File: sol/solution/0001-0100/0013. Roman to Integer/Solution.php
class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function romanToInt($s) {
        $map = ['I'=>1, 'V'=>5, 'X'=>10, 'L'=>50, 'C'=>100, 'D'=>500, 'M'=>1000];
        $res = 0;
        for ($i = 0; $i < strlen($s); $i++) {
            $val = $map[$s[$i]];
            $next = $i + 1 < strlen($s) ? $map[$s[$i + 1]] : 0;
            $res += ($val < $next) ? -$val : $val;
        }
        return $res;
    }
}